### DISCUSSION TITLE
Best practise for adding a new permission and granting it to existing users only?

### DISCUSSION THREAD

**Discussion by MCKLtech** [spatie/laravel-permission#2055] [Q&A]
I have a piece of functionality that all users had access to. However, I are now requiring permissions to use it e.g

`$user->can('add post')`

We want to grant all existing users this permission as a default, and only remove it if we remove it. Any users created after this deployment will not have this permission by default.

In other words, what is the best practice for creating new permission and granting it by default to all existing users?

**Comment by erikn69** [comment]
[gate before](https://spatie.be/docs/laravel-permission/v5/basic-usage/super-admin#content-gatebefore)
[gate after](https://spatie.be/docs/laravel-permission/v5/basic-usage/super-admin#content-gateafter)

**Reply by MCKLtech** [reply]
Thanks but I'm not sure I follow. No user has the 'add post' permission yet, but I want to grant it to all existing users of the app when I do add it so they don't lose access to a piece of functionality. 

Any user-created after this point in time will not have this permission by default.

**Reply by erikn69** [reply]
> [gate before](https://spatie.be/docs/laravel-permission/v5/basic-usage/super-admin#content-gatebefore)
 [gate after](https://spatie.be/docs/laravel-permission/v5/basic-usage/super-admin#content-gateafter)

You can make pass through validation with your own custom verification check, 
also https://github.com/spatie/laravel-permission/discussions/2021#discussioncomment-2193203
